Kernel/Syscall.cpp

```diff
@@ -15,60 +16,10 @@
 #include <Kernel/Sections.h>
 #include <Kernel/ThreadTracer.h>
 
-#if ARCH(X86_64)
-#    include <Kernel/Arch/x86_64/Interrupts.h>
-#endif
-
 namespace Kernel {
 
-extern "C" void syscall_handler(TrapFrame*) __attribute__((used));
-extern "C" void syscall_asm_entry();
-
-NEVER_INLINE NAKED void syscall_asm_entry()
-{
-    // clang-format off
-#if ARCH(X86_64)
-    asm(
-        "    pushq $0x0\n"
-        "    pushq %r15\n"
-        "    pushq %r14\n"
-        "    pushq %r13\n"
-        "    pushq %r12\n"
-        "    pushq %r11\n"
-        "    pushq %r10\n"
-        "    pushq %r9\n"
-        "    pushq %r8\n"
-        "    pushq %rax\n"
-        "    pushq %rcx\n"
-        "    pushq %rdx\n"
-        "    pushq %rbx\n"
-        "    pushq %rsp\n"
-        "    pushq %rbp\n"
-        "    pushq %rsi\n"
-        "    pushq %rdi\n"
-        "    pushq %rsp \n" /* set TrapFrame::regs */
-        "    subq $" __STRINGIFY(TRAP_FRAME_SIZE - 8) ", %rsp \n"
-        "    movq %rsp, %rdi \n"
-        "    cld\n"
-        "    call enter_trap_no_irq \n"
-        "    movq %rsp, %rdi \n"
-        "    call syscall_handler\n"
-        "    jmp common_trap_exit \n");
-#endif
-    // clang-format on
-}
-
 namespace Syscall {
 
-static ErrorOr<FlatPtr> handle(RegisterState&, FlatPtr function, FlatPtr arg1, FlatPtr arg2, FlatPtr arg3, FlatPtr arg4);
-
-UNMAP_AFTER_INIT void initialize()
-{
-#if ARCH(X86_64)
-    register_user_callable_interrupt_handler(syscall_vector, syscall_asm_entry);
```

Comment:
Is this function used anywhere else? if not it should be removed as well in this patch.

Reply:
Uhm, sorry but this function is not there in the PR, or maybe I'm missing something? You are referring to `initialize` right?

Reply:
I meant `register_user_callable_interrupt_handler`. It seems useful to check if this was the only call site

Reply:
Yes, that function is used:

```cpp
    register_user_callable_interrupt_handler(0x01, debug_asm_entry);
    register_user_callable_interrupt_handler(0x03, breakpoint_asm_entry);
```